File: solutions/r/prime-factors/2/prime-factors.R
prime_factors <- function(number) {
  if(number <= 1) return(c())
  if(number <= 3) return(c(number))
  index <- 1
  max <- floor(sqrt(number))+1
  primes <- rep(NA,max)
  divisor <- 2
  while(number > 1) {
    if(number%%divisor==0) {
      primes[index] <- divisor
      number <- number/divisor
      divisor <- 2
      index <- index+1
    } else {
      divisor <- divisor+1
    }
  }
  return(primes[!is.na(primes)])
}

File: solutions/julia/prime-factors/2/prime-factors.jl
function prime_factors(n)
    primes = []
    prospect = 3
    while n > 1
        while n%2 == 0
            push!(primes,2)
            n = n/2
        end
        if n <= 1 return primes end
        while n%prospect == 0
            push!(primes,prospect)
            n = n/prospect
        end
        if n <= 1 return primes end
        prospect += 2
    end
    return primes
end
